JahsieAult/homework.md

```diff
@@ -0,0 +1,164 @@
+
+### Homework #1
+(Due Monday 1/12)
+
+You need to turn in this homework by cloning this repo, making a new branch with your solutions, pushing to github, and issuing a pull request.
+For details, follow the instructions [here](http://portlandcodeschool.github.io/jse/2015/01/07/command-line-and-git-slides/#/14).
+
+Please also read the [learning objectives](objectives.md) for this week.
+
+---
+
+**1)** (_Difficulty: mixed_)
+Predict the output of each of the following expressions, and explain the rules which determine it.
+If the output depends on the value of x, identify the conditions when the expression will output true.  Assume the cases are independent, and x is reset to an unknown value before each.
+
+Some of these are tricky!  Don't trust your first instinct.  
+
+
+**a)** `"1" == 1`
+>True. The value of the string and the value of the number are equivalent. 
+
+**b)** `"1" === 1`
+>False. The values are the same but they are not the same AND of the same type. 
+
+**c)** `x == 'x'`
+>False. The first is a variable and the second is a string of 'x'. The variable could hold any value. This would be true if the value of x was "x"
+
+**d)** `x == (x+'')`
+>True. Adding an empty string concatenates nothing to the variable. This leaves it unchanged. 
```

Comment:
1d) There are a couple of weird edge cases here: `NaN`, which is not equal to anything (including itself!); and booleans, which get autoconverted to strings when added to strings.